backend/tests/v5/e2e/routes/teamspaces/roles.test.js

```diff
@@ -0,0 +1,214 @@
+		['teamspace does not exist', users.tsAdmin.apiKey, generateRandomString(), roleData, false, templates.teamspaceNotFound],
+		['user is not a member of the teamspace', users.nobody.apiKey, undefined, roleData, false, templates.teamspaceNotFound],
+		['user is not a teamspace admin', users.normalUser.apiKey, undefined, roleData, false, templates.notAuthorized],
+		['user is a ts admin and data is valid', users.tsAdmin.apiKey, undefined, roleData, true],
+		['data is invalid (missing name)', users.tsAdmin.apiKey, undefined, { ...roleData, name: undefined }, false, templates.invalidArguments],
+		['data is invalid (users with non ts members)', users.tsAdmin.apiKey, undefined, { ...roleData, users: [users.nobody.user] }, false, templates.invalidArguments],
+	])('Create role', (desc, key, ts, data, success, expectedRes) => {
+		test(`should ${success ? 'succeed if' : `fail with ${expectedRes.code}`} if ${desc}`, async () => {
+			const expectedStatus = success ? templates.ok.status : expectedRes.status;
+			const res = await agent.post(addRoleRoute(key, ts)).send(data).expect(expectedStatus);
+
+			if (success) {
+				expect(res.body?._id).not.toBeUndefined();
+				const { _id } = res.body;
+				const rolesRes = await agent.get(getRolesRoute(key, ts)).expect(templates.ok.status);
+				const role = rolesRes.body.roles.find((r) => r._id === _id);
+				expect(role).toEqual({ _id, ...data });
+			} else {
+				expect(res.body.code).toEqual(expectedRes.code);
+			}
+		});
+	});
+};
+
+const testUpdateRole = () => {
+	const basicData = generateBasicData();
+	const { users, teamspace, roles } = basicData;
+
+	beforeAll(async () => {
+		await setupData(basicData);
+	});
+
+	const roleData = { name: ServiceHelper.generateRandomString() };
+	const updateRoleRoute = (key, roleId, ts = teamspace) => `/v5/teamspaces/${ts}/roles/${roleId}${key ? `?key=${key}` : ''}`;
+
+	describe.each([
+		['user does not have a valid session', undefined, undefined, roles[0]._id, roleData, false, templates.notLoggedIn],
+		['teamspace does not exist', users.tsAdmin.apiKey, generateRandomString(), roles[0]._id, roleData, false, templates.teamspaceNotFound],
+		['user is not a member of the teamspace', users.nobody.apiKey, undefined, roles[0]._id, roleData, false, templates.teamspaceNotFound],
+		['user is not a teamspace admin', users.normalUser.apiKey, undefined, roles[0]._id, roleData, false, templates.notAuthorized],
+		['user is a ts admin and data is valid', users.tsAdmin.apiKey, undefined, roles[0]._id, roleData, true],
+		['role does not exist', users.tsAdmin.apiKey, undefined, generateRandomString(), roleData, false, templates.roleNotFound],
+		['data is invalid (empty body)', users.tsAdmin.apiKey, undefined, roles[0]._id, { }, false, templates.invalidArguments],
+		['data is invalid (users with non ts members)', users.tsAdmin.apiKey, undefined, roles[0]._id, { ...roleData, users: [users.nobody.user] }, false, templates.invalidArguments],
```

Comment:
Can we test:

- name already exists
- name is unchanged but submitted again (so detecting a name clash but it's the same role as the one we're trying to update)